extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java

```diff
@@ -73,6 +149,20 @@ public <T> Uni<T> executeWork(Context currentContext, Uni<T> uni, String workerN
         }
     }
 
+    private <T> Uni<T> runOnVirtualThread(Context currentContext, Uni<T> uni, Semaphore semaphore) {
+        return uni.runSubscriptionOn(command -> VIRTUAL_EXECUTOR_SUPPLIER.get().execute(command))
+                .onItemOrFailure().transformToUni((item, failure) -> {
+                    semaphore.release();
+                    return Uni.createFrom().emitter(emitter -> {
```

Comment:
Note for later, we have this construct multiple times and it's a bit allocation-heavy. 
We would need to think about it.

Might be a _sugar_ added in the Vert.x Mutiny bindings (WDYT @jponge).

Reply:
The whole `runSubscriptionOn` / `onItemOrFailure` / `transformToUni` dance?

Reply:
I was thinking just the "emit object on vert.x context"

Reply:
I'm still doing drafts of context-aware helpers 😉